ingestion/ingestion_server/ingestion_file_upload_rest_api.py

```diff
@@ -601,40 +607,36 @@ def delete_all():
     :param file_path:  Maps URL to files in templates directory.
     :return: Static file from directory along with data to display
     """
-    owner = current_user.get_id()
-    if not current_user.get_id():    # To run ingestion as stand-alone 
-        owner = "Alice"
+    owner = current_user.get_id() if current_user.get_id() else 'Alice'
     server_urls_instance = ServerUrls().get_instance()
     db_core = MainDb.get_core_db_instance()
     fs = gridfs.GridFSBucket(db_core.get_magen_mdb())
-    files_list = fs.find({"metadata.owner": owner})
+    files_list = fs.find({'metadata.owner': owner})
     resp = []
     try:
         for f in files_list:
             if 'type' not in f.metadata:
                 get_return_obj = RestClientApis.http_get_and_check_success(
                     server_urls_instance.key_server_single_asset_url.format(f.metadata['asset_uuid']))
 
-                if get_return_obj.success:
-                    key_id = get_return_obj.to_dict()['json']['response']['key']['key_id']
-                    key_return_obj = RestClientApis.http_delete_and_check_success(
-                        server_urls_instance.key_server_asset_keys_keys_key_url + key_id + "/")
-
-                    asset_return_obj = RestClientApis.http_delete_and_get_check(
-                        server_urls_instance.ingestion_server_single_asset_url.format(f.metadata['asset_uuid']))
-                    if key_return_obj.success and asset_return_obj.success:
-                        resp.append("success")
-                    else:
-                        raise Exception("Error deleting Key/Asset")
-                else:
+                if not get_return_obj.success:
                     raise Exception("Error Key Server Problem")
+                key_id = get_return_obj.to_dict()['json']['response']['key']['key_id']
+                key_return_obj = RestClientApis.http_delete_and_check_success(
+                    server_urls_instance.key_server_asset_keys_keys_key_url + key_id + "/")
+
+                asset_return_obj = RestClientApis.http_delete_and_get_check(
+                    server_urls_instance.ingestion_server_single_asset_url.format(f.metadata['asset_uuid']))
+                if not key_return_obj.success or not asset_return_obj.success:
+                    raise Exception("Error deleting Key/Asset")
+                resp.append("success")
+
             elif 'type' in f.metadata:
                 asset_return_obj = RestClientApis.http_delete_and_get_check(
                     server_urls_instance.ingestion_server_single_asset_url.format(f.metadata['asset_uuid']))
-                if asset_return_obj.success:
-                    resp.append("success")
-                else:
+                if not asset_return_obj.success:
                     raise Exception("Error deleting Asset")
```

Comment:
instead of raising an exception we probably could return an error object and return it directly to the user?

exception handling works through exception stack and is slower then normal return - it can affect our performance. We should use Exceptions, but no too greedy.